### PR TITLE
(hopefully) improvements to use of scope and kinesis module

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request
so, scope and kinesis are the only things in the game (other than the body zone selector) which use the function MouseMove. this tracks every mouse movement, which meant we had to stuff a cooldown on it to not calculate a ton of useless stuff. this time can misalign if you move your mouse fast, not registering at all, as well as not working out with the 0.2 second processing time of the things handling it (the scope component and kinesis module)
instead of doing that, we are now keeping the mouse parameters as a variable, which we update with every mousemove to the current parameters. then we handle the calculations right as we need them (in the kinesis/scope) module, rather than relying on mousemove cooldowns, this should hopefully feel way better

## Why It's Good For The Game
😁 

## Changelog
:cl:
qol: sniper scopes and kinesis module should feel better to use
/:cl:

### CODE REVIEW

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/datums/view.dm [hunk 1]
```diff
@@ -38,8 +38,8 @@
 		assertFormat()
 	else
 		resetFormat()
-	var/datum/hud/our_hud = chief?.mob?.hud_used
-	our_hud.view_audit_buttons() // Make sure our hud's buttons are in our new size
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/datums/view.dm:42]: it became a signal proc

[the remark this answers is not in the document]
Comment (Fikou) [reply code/datums/view.dm:42]: im dumb as fuck

File: code/datums/components/scope.dm [hunk 2]
```diff
@@ -110,9 +113,12 @@
 	tracker = user.overlay_fullscreen("scope", /atom/movable/screen/fullscreen/scope, 0)
 	tracker.range_modifier = range_modifier
 	tracker.marksman = user
+	tracker.view_list = getviewsize(user.client.view)
 	tracker.RegisterSignal(user, COMSIG_MOVABLE_MOVED, /atom/movable/screen/fullscreen/scope.proc/on_move)
+	tracker.RegisterSignal(user, COMSIG_VIEWDATA_UPDATE, /atom/movable/screen/fullscreen/scope.proc/on_viewdata_update)
+	tracker.calculate_params()
 	RegisterSignal(user, COMSIG_MOB_SWAP_HANDS, .proc/stop_zooming)
-	START_PROCESSING(SSfastprocess, src)
+	START_PROCESSING(SSprojectiles, src)
```
Comment (LemonInTheDark) [inline code/datums/components/scope.dm:121]: is this just for the speed

File: code/modules/mod/modules/module_kinesis.dm [hunk 3]
```diff
@@ -64,15 +66,18 @@
...
 	grabbed_atom.add_overlay(kinesis_icon)
 	kinesis_beam = mod.wearer.Beam(grabbed_atom, "kinesis")
 	kinesis_catcher = mod.wearer.overlay_fullscreen("kinesis", /atom/movable/screen/fullscreen/kinesis, 0)
 	kinesis_catcher.kinesis_user = mod.wearer
+	kinesis_catcher.view_list = getviewsize(mod.wearer.client.view)
 	kinesis_catcher.RegisterSignal(mod.wearer, COMSIG_MOVABLE_MOVED, /atom/movable/screen/fullscreen/kinesis.proc/on_move)
+	kinesis_catcher.RegisterSignal(mod.wearer, COMSIG_VIEWDATA_UPDATE, /atom/movable/screen/fullscreen/kinesis.proc/on_viewdata_update)
+	kinesis_catcher.calculate_params()
 	soundloop.start()
+	START_PROCESSING(SSfastprocess, src)
```
Comment (LemonInTheDark) [inline code/modules/mod/modules/module_kinesis.dm:80]: if so, why is this not also on it? overhead?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/__DEFINES/dcs/signals/signals_mob/signals_mob_main.dm [hunk 4]
```diff
@@ -46,6 +46,8 @@
 #define COMSIG_MOB_RESET_PERSPECTIVE "mob_reset_perspective"
 /// from base of /client/proc/set_eye() : (atom/old_eye, atom/new_eye)
 #define COMSIG_CLIENT_SET_EYE "client_set_eye"
+/// from base of /datum/view_data/proc/afterViewChange() : (width, height)
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/__DEFINES/dcs/signals/signals_mob/signals_mob_main.dm:49]: i use the view as a list

File: code/datums/components/scope.dm (hunk 2 above)
Comment (Fikou) [reply code/datums/components/scope.dm:121]: yes

File: code/modules/mod/modules/module_kinesis.dm (hunk 3 above)
Comment (Fikou) [reply code/modules/mod/modules/module_kinesis.dm:80]: doesnt matter here as much

**Pull request merged**